project-koku/koku
# [COST-5034] - Use json_query for OCP on Azure sql tag matching

## Jira Ticket

[COST-5034](https://issues.redhat.com/browse/COST-5034)

## Description

This change will tweak our special case tag matching during OCP on Azure trino SQL to avoid false positives on generic string matching.

## Testing

Manually run some trino sql:
1. create table azure_tag_test (tags varchar, matched_tag varchar);
2. insert into azure_tag_test (tags, matched_tag) values ('{"environment":"test","namespace":"fake-namespace"}', '');
3. create table ocp_name_test (namespace varchar);
4. insert into ocp_name_test (namespace) values ('test');
5. Try to match the data!: SELECT azure.tags, ocp.namespace, ((strpos(lower(azure.tags), 'namespace') != 0 AND strpos(lower(azure.tags), lower(ocp.namespace)) != 0)) AS match FROM azure_tag_test AS azure CROSS JOIN ocp_name_test AS ocp;
                                             	tags                                             | namespace | match
-------------------------------------------------------------------------------------------------------+-----------+-------
 {"environment":"test","namespace":"fake-namespace"} | test              | true  

The above (OLD method) will match fine

Whereas the below (New) method fails to match (expected)

SELECT azure.tags, ocp.namespace, (json_query(azure.tags, 'strict $.namespace' OMIT QUOTES) = ocp.namespace) AS match FROM azure_tag_test AS azure CROSS JOIN ocp_name_test AS ocp;

 {"environment":"test","namespace":"fake-namespace"} | test              | false

 You could also test a correct match by inserting the following: 
insert into ocp_name_test (namespace) values ('fake-namespace');
Then run the above sql again


## Release Notes

```markdown
* [COST-5034](https://issues.redhat.com/browse/COST-5034) Fix ocp on azure tag matching SQL
```
